### PR TITLE
Add an argument to omit row_sparse_push for end to end benchmarking

### PR DESCRIPTION
**Pull request by anirudh2290**
@eric-haibin-lin @reminisce

### CODE REVIEW

**Comment by anirudh2290** [comment]
The purpose of skipping row_sparse_push is to calculate only communication cost which is associated with row_sparse_pull. We used this to obtain the communication cost documented here : https://github.com/anirudh2290/mxnet/wiki/Sparse-Tensor-Multimachine-End-to-End-benchmarking-results#multi-machine-communication-cost-only-row_sparse_pull-dummy-iterator-used-for-io-kvstore--dist_async

**Comment by anirudh2290** [comment]
@piiswrong can you please review and merge.

File: benchmark/python/sparse/sparse_end2end.py [hunk 1]
```diff
@@ -57,13 +57,15 @@
...
 
 def get_libsvm_data(data_dir, data_name, url, data_origin_name):
     if not os.path.isdir(data_dir):
         os.system("mkdir " + data_dir)
     os.chdir(data_dir)
-    if (not os.path.exists(data_name)):
+    if not os.path.exists(data_name):
         import urllib
         zippath = os.path.join(data_dir, data_origin_name)
         urllib.urlretrieve(url, zippath)
```
Comment (szha) [inline benchmark/python/sparse/sparse_end2end.py]: Use bz2 instead of relying on the bzip2 command in the os. https://docs.python.org/2/library/bz2.html

Comment (szha) [reply benchmark/python/sparse/sparse_end2end.py]: for portability

Comment (anirudh2290) [reply benchmark/python/sparse/sparse_end2end.py]: Moved the get_bz2_data to test_utils and reused it in tests and benchmarking scripts.

**Pull request merged**